app/models/user.rb

```diff
@@ -106,12 +105,16 @@ def self.create_or_update(json_user)
     wca_create_or_update(json_user)
   end
 
-  def try_associate_subscriptions
-    unless wca_id.blank?
-      Subscription.userless.where(wca_id: wca_id).update(user_id: id)
-    end
-    Subscription.userless.where("concat(lower(firstname), ' ', lower(name)) = ?", name.downcase).update(user_id: id)
+  def add_charge(amount, stripe_charge_id)
+    charges.create!(
+      amount: amount,
+      stripe_charge_id: stripe_charge_id,
+    )
   end
 
-  private :try_associate_subscriptions
+  def add_subscription(stripe_charge_id)
+    payed_at = Charge.where(user_id: id).last.created_at
```

Comment:
Hum that's not really good, in some scenario the user might subscribe twice at the same time, and you're not guaranteed that `last` will return the correct charge.
I'd either pass a "Charge" object to this method (that you can accurately get because `add_charge` returns the value of `charges.create!` which returns the created object), or simply use only one object (merge `Charge`'s attributes into `Subscription`).

Reply:
In line with your other comment, probably just merging them is best. I'll try that!